main.tf

```diff
@@ -1,13 +1,192 @@
+  }
+
+  task_invocation_parameters {
+    run_command_parameters {
+      parameter {
+        name   = "Operation"
+        values = ["Install"]
+      }
+      parameter {
+        name   = "RebootOption"
+        values = [var.reboot_option]
+      }
+      output_s3_bucket     = local.bucket_id
+      output_s3_key_prefix = var.s3_bucket_prefix_install_logs
+      service_role_arn     = var.role_arn_for_notification
+
+      dynamic "notification_config" {
+        for_each = var.enable_notification_install ? [1] : []
+        content {
+          notification_arn    = var.notification_arn
+          notification_events = var.notification_events
+          notification_type   = var.notification_type
+        }
+      }
+    }
+  }
+}
+
+resource "aws_ssm_maintenance_window_target" "target_install" {
+  count         = local.enabled ? 1 : 0
+  window_id     = aws_ssm_maintenance_window.install_window[0].id
+  resource_type = "INSTANCE"
+
+  dynamic "targets" {
+    for_each = toset(var.install_maintenance_windows_targets)
+    content {
+      key    = targets.value.key
+      values = targets.value.values
+    }
+  }
+
+  dynamic "targets" {
+    for_each = length(var.install_maintenance_windows_targets) == 0 ? [1] : []
+    content {
+      key    = "tag:Patch Group"
+      values = var.install_patch_groups
+    }
+  }
+}
+
+# Patch Baselines
+resource "aws_ssm_patch_baseline" "baseline" {
+  count            = local.enabled ? 1 : 0
+  name             = "${module.this.id}-${var.operating_system}"
+  description      = "Amazon linux 2 baseline"
+  operating_system = var.operating_system
+
+  approved_patches                  = var.approved_patches
+  rejected_patches                  = var.rejected_patches
+  approved_patches_compliance_level = var.approved_patches_compliance_level
+
+  dynamic "approval_rule" {
+    for_each = toset(var.patch_baseline_approval_rules)
+    content {
+
+      approve_after_days  = approval_rule.value.approve_after_days
+      compliance_level    = approval_rule.value.compliance_level
+      enable_non_security = approval_rule.value.enable_non_security
+
+      # https://docs.aws.amazon.com/cli/latest/reference/ssm/describe-patch-properties.html
+      dynamic "patch_filter" {
+        for_each = approval_rule.value.patch_baseline_filters
+
+        content {
+          key    = patch_filter.value.name
+          values = patch_filter.value.values
+        }
+      }
+    }
+  }
+  tags = var.tags
+}
+
+resource "aws_ssm_patch_group" "install_patchgroup" {
+  count       = (local.enabled ? 1 : 0) * length(var.scan_patch_groups)
+  baseline_id = aws_ssm_patch_baseline.baseline[0].id
+  patch_group = element(var.install_patch_groups, count.index)
+}
+
+resource "aws_ssm_patch_group" "scan_patchgroup" {
+  count       = (local.enabled ? 1 : 0) * length(var.scan_patch_groups)
+  baseline_id = aws_ssm_patch_baseline.baseline[0].id
+  patch_group = element(var.scan_patch_groups, count.index)
 }
```

Comment:
These two are the same, you prob need to update/fix the `install_patchgroup` one


```
resource "aws_ssm_patch_group" "install_patchgroup" {
  count       = (local.enabled ? 1 : 0) * length(var.scan_patch_groups)
  baseline_id = aws_ssm_patch_baseline.baseline[0].id
  patch_group = element(var.install_patch_groups, count.index)
}

resource "aws_ssm_patch_group" "scan_patchgroup" {
  count       = (local.enabled ? 1 : 0) * length(var.scan_patch_groups)
  baseline_id = aws_ssm_patch_baseline.baseline[0].id
  patch_group = element(var.scan_patch_groups, count.index)
}
```